rtos/TARGET_CORTEX/mbed_rtx_idle.cpp

```diff
@@ -90,6 +90,13 @@ uint32_t OS_Tick_GetInterval (void) {
 static void default_idle_hook(void)
 {
     uint32_t ticks_to_sleep = osKernelSuspend();
+    const bool block_deep_sleep = ticks_to_sleep <= MBED_CONF_TARGET_DEEP_SLEEP_LATENCY;
+
+    if (block_deep_sleep) {
+        sleep_manager_lock_deep_sleep();
+    } else {
+        ticks_to_sleep -= MBED_CONF_TARGET_DEEP_SLEEP_LATENCY;
```

Comment:
@c1728p9 - was discussing this with @sg-, and it occurred to me that this is being unduly pessimistic - you're subtracting the deep sleep latency from the wakeup time _even if you're not going to be deep sleeping anyway_, so in shallow operation there will be lots of unnecessary double firing.

May have to make the critical section wrapper bigger to synchronise a read of `sleep_manager_can_deep_sleep`. Maybe we could partially compensate by stripping the `sleep_manager_lock_deep_sleep_internal()` down - it doesn't need its critical section.